### PR TITLE
MOSIP-44844

### PR DESCRIPTION
**Pull request by dhanendra06**
## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed packet creation date handling in UIN generation when missing from schema.
  * Adjusted transaction handling in packet classification to alter transactional boundaries.

* **Performance Improvements**
  * Optimized packet upload to reuse unzipped data and run ZIP/JSON uploads concurrently.
  * Improved parallelism and async handling in workflow processing for faster calculations.

* **Tests**
  * Expanded tests for packet creation date scenarios and related edge cases.

* **Chores**
  * Updated object store dependency version.

### CODE REVIEW

**Comment by ashok-ksharma** [comment]
@coderabbitai review

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java [hunk 1]
```diff
@@ -506,39 +512,70 @@ private MessageDTO uploadPacket(SyncRegistrationEntity regEntity, InternalRegist
...
                     byte[] bytearray = IOUtils.toByteArray(entry.getValue());
-                    String jsonString = new String(bytearray);
-                    LinkedHashMap<String, Object> currentIdMap = (LinkedHashMap<String, Object>) mapper.readValue(jsonString, LinkedHashMap.class);
-                    String objStoreKey = isIterationAdditionEnabled ?
-                            getFinalKey(regEntity, entry.getKey().replace(JSON, ""), object)
-                            :
-                            entry.getKey().replace(JSON, "");
-                    objectStoreAdapter.addObjectMetaData(packetManagerAccount, registrationId,
-                            null, null, objStoreKey, currentIdMap);
+                    LinkedHashMap<String, Object> currentIdMap = (LinkedHashMap<String, Object>) mapper.readValue(new String(bytearray), LinkedHashMap.class);
```
[the remark this answers is not in the document]
Comment (ashok-ksharma) [reply registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java:553]: We have already tested the same with new String(bytearray) and hence not taking the code change.

File: registration-processor/pre-processor/registration-processor-securezone-notification-stage/src/main/java/io/mosip/registration/processor/securezone/notification/stage/SecurezoneNotificationStage.java [hunk 2]
```diff
@@ -212,15 +212,23 @@ public MessageDTO process(MessageDTO messageDTO) {
 		messageDTO.setIsValid(Boolean.FALSE);
 		boolean isTransactionSuccessful = false;
 		try {
+			long getRegistrationStatusStartTime = System.currentTimeMillis();
 			registrationStatusDto = registrationStatusService.getRegistrationStatus(messageDTO.getRid(),
 					messageDTO.getReg_type(), messageDTO.getIteration(), messageDTO.getWorkflowInstanceId());
+			regProcLogger.info(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					messageDTO.getRid(), "Perf_SecurezoneNotificationStage_process_getRegistrationStatus"
+							+ " -- elapsedTime: " + (System.currentTimeMillis() - getRegistrationStatusStartTime) + "ms");
```
[the remark this answers is not in the document]
Comment (ashok-ksharma) [reply registration-processor/pre-processor/registration-processor-securezone-notification-stage/src/main/java/io/mosip/registration/processor/securezone/notification/stage/SecurezoneNotificationStage.java]: Removed the unnecessary loggers.

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java [hunk 3]
```diff
@@ -506,39 +512,70 @@ private MessageDTO uploadPacket(SyncRegistrationEntity regEntity, InternalRegist
...
+                    final String objStoreKey = isIterationAdditionEnabled ?
+                            getFinalKey(regEntity, entry.getKey().replace(JSON, ""), object, additionalInfoRequestDto)
+                            : entry.getKey().replace(JSON, "");
+                    jsonFutures.add(CompletableFuture.runAsync(() ->
+                            objectStoreAdapter.addObjectMetaData(packetManagerAccount, registrationId,
+                                    null, null, objStoreKey, currentIdMap), uploadExecutor));
```
[the remark this answers is not in the document]
Comment (ashok-ksharma) [reply registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java:559]: Previous code also does not validate the same hence we are in sync with older code.

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java [hunk 4]
```diff
@@ -576,15 +613,15 @@ private byte[] getPakcetFromDMZ(String packetId, String registrationId) throws A
...
-        	}
+            if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.DMZ_SERVER)) {
+                packet = (byte[]) restClient.getApi(ApiName.NGINXDMZURL, pathSegment, "", null, byte[].class);
+            }
+            else if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.OBJECT_STORE)) {
+                packet=IOUtils.toByteArray(objectStoreAdapter.getObject(landingZoneAccount, registrationId, null, null, packetId));
+                if(packet==null) {
+                    throw new ObjectStoreNotAccessibleException("Failed to get packet : " +packetId);
+                }
+            }
```
[the remark this answers is not in the document]
Comment (ashok-ksharma) [reply registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java:630]: Previous code also does not validate the same hence we are in sync with older code.

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 5]
```diff
@@ -260,38 +262,180 @@ public MessageDTO process(MessageDTO object) {
...
+							throw new RuntimeException(e);
+						}
+					}, virtualThreadExecutor);
+
+			// Collect all results
+			InternalRegistrationStatusDto registrationStatusDto = registrationStatusFuture.get();
+			Map<String, String> metaInfoMap = metaInfoFuture.get();
+			BiometricRecord biometricRecord = biometricsFuture.get();
+			Map<String, String> fieldTypeMap = fieldTypeMapFuture.get();
+			Map<String, String> fieldMap = fieldMapFuture.get();
```
[the remark this answers is not in the document]
Comment (ashok-ksharma) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java:349]: This is fine as we are not using the timeout in other places in the code. We will check later if it is required.

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashok-ksharma) [reply registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java:553]: This is in sync with older code hence no change is required.

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java [hunk 6]
```diff
@@ -593,7 +636,7 @@ else if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.OBJECT_STORE))
             } else
                 throw e;
         } catch(ObjectStoreAdapterException e) {
-        	throw e;
+            throw e;
```
[the remark this answers is not in the document]
Comment (ashok-ksharma) [reply registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java:639]: This is in sync with older code hence no change is required.

**Review by ashok-ksharma — APPROVED** [review]
